packages/adblocker/src/filters/network.ts

```diff
@@ -400,18 +397,169 @@ function compileRegex(
   return new RegExp(filter);
 }
 
-export function findLastIndexOfUnescapedCharacter(text: string, character: string) {
-  let lastIndex = text.lastIndexOf(character);
+/**
+ * Collects a filter option key until the function sees the special character.
+ * This function will stop iterating over the given string if it sees equal sign or comma sign.
+ * If there's an equal sign, it means that we'll see the value.
+ * Otherwise, if there's a comma sign, it means that the option doesn't have any values.
+ * Note that this function doesn't respect the escaping sign.
+ */
+function getFilterOptionKey(line: string, pos: number, end: number) {
+  let code: number;
+  let value = '';
+
+  for (; pos < end; pos++) {
+    code = line.charCodeAt(pos);
+
+    if (code === 61 /* '=' */ || code === 44 /* ',' */) {
+      break;
+    }
+
+    value += line.charAt(pos);
+  }
+
+  return [pos, value] as const;
+}
+
+/**
+ * Collects a filter option value until the function sees the special character.
+ * This function respects the escaping characters, so we can safely collect the full value
+ * including the special characters which are not allowed normally.
+ * This function will stop if it sees a comma sign.
+ */
+function getFilterOptionValue(line: string, pos: number, end: number): [number, string] {
+  let code: number;
+  let value = '';
+
+  for (; pos < end; pos++) {
+    code = line.charCodeAt(pos);
+
+    if (code === 92 /* '\\' */) {
+      value += line.charAt(++pos);
+    } else if (code === 44 /* ',' */) {
+      break;
+    } else {
+      value += line.charAt(pos);
+    }
+  }
+
+  return [pos, value];
+}
+
+/**
+ * Collects a filter option value of the replace modifier.
+ * This function respects the escaping character with the allowed characters of the replace modifier.
+ * In the replace modifier, it can include the any sign allowed in the regular expression.
+ * Therefore, a comma sign can interfere the `getFilterOptionValue` function.
+ * This function will not stop unless it collects the all of parts of the replace modifier option value.
+ */
+export function getFilterReplaceOptionValue(
+  line: string,
+  pos: number,
+  end: number,
+): [number, string[]] {
+  // Try to fast exit if the first character is an unexpected character.
+  if (line.charCodeAt(pos++) !== 47 /* '/' */) {
+    return [end, []];
+  }
+
+  const parts = ['', '', ''];
+
+  let code: number;
+  let slashes = 0;
+
+  for (; pos < end; pos++) {
+    code = line.charCodeAt(pos);
+
+    if (code === 92 /* '\\' */) {
+      parts[slashes] += '\\' + line.charAt(++pos);
+    } else if (code === 47 /* '/' */) {
+      if (++slashes === 2) {
+        // Skip the last slash character
+        // Since we saw 3 slashes in total, it means that the option value should be closed here.
```

Comment:
I assume this here should mean 2 instead of 3?!
```suggestion
        // Since we saw 2 slashes in total, it means that the option value should be closed here.
```

On the other hand, I wonder what happens if the regular expression looks like that? (Quoting `/` not as `\/`, but as `[/]`):

```
/[/]/_/

Reply:
In this case, the slash in the matching regexp should be escaped by the syntax design.